#### Import Packages

In [1]:
import numpy as np
import pandas as pd 

#### Import CSV files into pandas dataframes

In [2]:
data_items = pd.read_csv('Resources/Items.csv')
print('Shape of dataset is', data_items.shape)
data_items.head()

Shape of dataset is (720, 10)


asin     brand                                              title  \
0  B0000SX2UC       NaN  Dual-Band / Tri-Mode Sprint PCS Phone w/ Voice...   
1  B0009N5L7K  Motorola                                Motorola I265 phone   
2  B000SKTZ0S  Motorola  MOTOROLA C168i AT&T CINGULAR PREPAID GOPHONE C...   
3  B001AO4OUC  Motorola              Motorola i335 Cell Phone Boost Mobile   
4  B001DCJAJG  Motorola      Motorola V365 no contract cellular phone AT&T   

                                                 url  \
0  https://www.amazon.com/Dual-Band-Tri-Mode-Acti...   
1  https://www.amazon.com/Motorola-i265-I265-phon...   
2  https://www.amazon.com/MOTOROLA-C168i-CINGULAR...   
3  https://www.amazon.com/Motorola-i335-Phone-Boo...   
4  https://www.amazon.com/Motorola-V365-contract-...   

                                               image  rating  \
0  https://m.media-amazon.com/images/I/2143EBQ210...     3.0   
1  https://m.media-amazon.com/images/I/419WBAVDAR...     3.0   
2  https://m.media-amazon.com/images/I/71b+q3ydkI...     2.7   
3  https://m.media-amazon.com/images/I/710UO8gdT+...     3.3   
4  https://m.media-amazon.com/images/I/61LYNCVrrK...     3.1   

                                           reviewUrl  totalReviews   price  \
0  https://www.amazon.com/product-reviews/B0000SX2UC            14    0.00   
1  https://www.amazon.com/product-reviews/B0009N5L7K             7   49.95   
2  https://www.amazon.com/product-reviews/B000SKTZ0S            22   99.99   
3  https://www.amazon.com/product-reviews/B001AO4OUC            21    0.00   
4  https://www.amazon.com/product-reviews/B001DCJAJG            12  149.99   

   originalPrice  
0            0.0  
1            0.0  
2            0.0  
3            0.0  
4            0.0

In [3]:
data_reviews = pd.read_csv('Resources/Reviews.csv')
print('Shape of dataset is', data_reviews.shape)
#Round off average rating to 1 decimal place
data_reviews=data_reviews.round({'rating':1})
data_reviews.head()


Shape of dataset is (67986, 8)


asin           name  rating       date  verified  \
0  B0000SX2UC          Janet       3  11-Oct-05     False   
1  B0000SX2UC     Luke Wyatt       1   7-Jan-04     False   
2  B0000SX2UC         Brooke       5  30-Dec-03     False   
3  B0000SX2UC  amy m. teague       3  18-Mar-04     False   
4  B0000SX2UC  tristazbimmer       4  28-Aug-05     False   

                                          title  \
0                   Def not best, but not worst   
1                   Text Messaging Doesn't Work   
2                               Love This Phone   
3                       Love the Phone, BUT...!   
4  Great phone service and options, lousy case!   

                                                body  helpfulVotes  
0  I had the Samsung A600 for awhile which is abs...           1.0  
1  Due to a software issue between Nokia and Spri...          17.0  
2  This is a great, reliable phone. I also purcha...           5.0  
3  I love the phone and all, because I really did...           1.0  
4  The phone has been great for every purpose it ...           1.0

Since the data file Amazon_Unlocked_Mobile is large, we only get the required columns ProductName and Brand Name that are necessary for the analysis:

In [4]:
data_Amazon_Unlocked_Mobile = pd.read_csv('Resources/Amazon_Unlocked_Mobile.csv', usecols=[0,1])
print('Shape of dataset is', data_Amazon_Unlocked_Mobile.shape)
data_Amazon_Unlocked_Mobile.head()

Shape of dataset is (413840, 2)


Product Name Brand Name
0  "CLEAR CLEAN ESN" Sprint EPIC 4G Galaxy SPH-D7...    Samsung
1  "CLEAR CLEAN ESN" Sprint EPIC 4G Galaxy SPH-D7...    Samsung
2  "CLEAR CLEAN ESN" Sprint EPIC 4G Galaxy SPH-D7...    Samsung
3  "CLEAR CLEAN ESN" Sprint EPIC 4G Galaxy SPH-D7...    Samsung
4  "CLEAR CLEAN ESN" Sprint EPIC 4G Galaxy SPH-D7...    Samsung

In [5]:
data_reviews.shape

(67986, 8)

### Data exploration and cleanup

#### Dropping the data with NaN records in the datasets

In [6]:
data_reviews.dropna(inplace=True)
data_reviews.shape

(27206, 8)

In [7]:
data_Amazon_Unlocked_Mobile.dropna(inplace=True)
data_Amazon_Unlocked_Mobile.shape

(348669, 2)

In [8]:
data_items.dropna(inplace=True)
data_items.shape

(716, 10)

In [9]:
data_reviews

asin           name  rating       date  verified  \
0      B0000SX2UC          Janet       3  11-Oct-05     False   
1      B0000SX2UC     Luke Wyatt       1   7-Jan-04     False   
2      B0000SX2UC         Brooke       5  30-Dec-03     False   
3      B0000SX2UC  amy m. teague       3  18-Mar-04     False   
4      B0000SX2UC  tristazbimmer       4  28-Aug-05     False   
...           ...            ...     ...        ...       ...   
67978  B081H6STQQ      Rock Edge       5  16-Jul-19     False   
67979  B081H6STQQ   Cindy Cowles       3  17-Jul-19     False   
67980  B081H6STQQ    Los Kositos       5   7-Oct-19     False   
67981  B081H6STQQ          jande       5  16-Aug-19     False   
67982  B081H6STQQ        2cool4u       5  14-Sep-19     False   

                                                   title  \
0                            Def not best, but not worst   
1                            Text Messaging Doesn't Work   
2                                        Love This Phone   
3                                Love the Phone, BUT...!   
4           Great phone service and options, lousy case!   
...                                                  ...   
67978                           Candy bar phone is back!   
67979                                     Updated review   
67980               From iPhone to Android and loving it   
67981  Awesome Phone, but finger scanner is a big mis...   
67982                                    Simply Amazing!   

                                                    body  helpfulVotes  
0      I had the Samsung A600 for awhile which is abs...           1.0  
1      Due to a software issue between Nokia and Spri...          17.0  
2      This is a great, reliable phone. I also purcha...           5.0  
3      I love the phone and all, because I really did...           1.0  
4      The phone has been great for every purpose it ...           1.0  
...                                                  ...           ...  
67978  Update 8/14/19 5 stars now! I've been using th...          12.0  
67979  Update: If you like sending and receiving pict...           2.0  
67980  I love my new phone. I've been dying to do a r...           1.0  
67981  I love the camera on this phone. The screen is...           1.0  
67982  I've been an Xperia user for several years and...           1.0  

[27206 rows x 8 columns]

Understanding the distribution of verified rows:

In [10]:
data_reviews['verified'].value_counts()

True     23154
False     4052
Name: verified, dtype: int64

Since the non-verified rows are not authentic, we can drop those records and keep only the verified ones.

In [11]:
data_reviews = data_reviews[data_reviews['verified'] == True]

In [12]:
data_reviews

asin                 name  rating       date  verified  \
43     B001AO4OUC            Elizabeth       4  12-May-09      True   
44     B001AO4OUC  Ron Malinowski, Jr.       1  19-Nov-12      True   
49     B001AO4OUC                PENNY       4  25-Nov-12      True   
53     B001AO4OUC  Anthony L. Savarese       1  18-Oct-09      True   
64     B001DCJAJG      Kindle Customer       4  22-Jun-14      True   
...           ...                  ...     ...        ...       ...   
67970  B081H6STQQ           SoundFreak       1  23-Jul-19      True   
67971  B081H6STQQ             Bialynia       5  14-Jul-19      True   
67972  B081H6STQQ              Aucifer       5  17-Jul-19      True   
67973  B081H6STQQ             Gclancer       5  14-Aug-19      True   
67974  B081H6STQQ               Pierre       3   3-Oct-19      True   

                                                   title  \
43                                    tough little phone   
44            Boost Mobile no Longer allows IDEN service   
49                                              SIM CARD   
53                                                  Cell   
64                              Good phone...good price.   
...                                                  ...   
67970  DO NOT BUY ( Especially if using TMobile ) !!!...   
67971                               Great phone (for me)   
67972                        Best Screen on a phone 2019   
67973                           Much improved Sony phone   
67974                                  I'm not impressed   

                                                    body  helpfulVotes  
43     This phone isn't kidding when it says military...           1.0  
44     I bought this phone and found out IDEN phones ...           2.0  
49     PHONE DID NOT COME WITH A SIM CARD....INFORMAT...           1.0  
53     Wouldn't know anything about the cell phone I ...           1.0  
64     This was a replacement for the one my mom had ...           2.0  
...                                                  ...           ...  
67970  update Nov 4, 2019 This phone is crap if you a...           7.0  
67971  All the manufacturer's make great phones, so p...           6.0  
67972  Coming from sony xperia XZ Premium is a nice u...           3.0  
67973  Beautiful phone, gorgeous display, fast near s...           2.0  
67974  For this price you get a good display and a ma...           1.0  

[23154 rows x 8 columns]

In [13]:
#verification:
data_reviews['verified'].value_counts()

True    23154
Name: verified, dtype: int64

Reviewing the price of items:

In [14]:
data_items['price'].value_counts()

0.00      123
99.99       8
249.99      8
129.99      7
399.99      7
         ... 
139.97      1
351.87      1
309.95      1
275.99      1
354.99      1
Name: price, Length: 451, dtype: int64

We can see that there are 123 items with price 0 which do not make sense. So, dropping these items with no price.

In [15]:
data_items = data_items[data_items['price'] != 0]

In [16]:
#verification:
data_items['price'].value_counts()

99.99     8
249.99    8
129.99    7
499.99    7
399.99    7
         ..
310.99    1
139.97    1
351.87    1
309.95    1
174.95    1
Name: price, Length: 450, dtype: int64

In [17]:
data_Amazon_Unlocked_Mobile

Product Name Brand Name
0       "CLEAR CLEAN ESN" Sprint EPIC 4G Galaxy SPH-D7...    Samsung
1       "CLEAR CLEAN ESN" Sprint EPIC 4G Galaxy SPH-D7...    Samsung
2       "CLEAR CLEAN ESN" Sprint EPIC 4G Galaxy SPH-D7...    Samsung
3       "CLEAR CLEAN ESN" Sprint EPIC 4G Galaxy SPH-D7...    Samsung
4       "CLEAR CLEAN ESN" Sprint EPIC 4G Galaxy SPH-D7...    Samsung
...                                                   ...        ...
413835  Samsung Convoy U640 Phone for Verizon Wireless...    Samsung
413836  Samsung Convoy U640 Phone for Verizon Wireless...    Samsung
413837  Samsung Convoy U640 Phone for Verizon Wireless...    Samsung
413838  Samsung Convoy U640 Phone for Verizon Wireless...    Samsung
413839  Samsung Convoy U640 Phone for Verizon Wireless...    Samsung

[348669 rows x 2 columns]

#### Aggregation

Getting the count of possible products for each brand to map to the other 2 datasets:

In [18]:
data_Amazon_Unlocked_New = data_Amazon_Unlocked_Mobile.groupby('Brand Name')['Product Name'].nunique().reset_index(name = 'Product Count/Brand')

#### Renaming the columns and keeping only the required columns from all datasets:

In [19]:
data_items = data_items[['asin','brand','title','price','originalPrice','rating']].rename(columns={'asin': 'ItemId', 'brand': 'Brand Name','title': 'ItemName','originalPrice': 'OriginalPrice','price':'ItemPrice','rating':'Rating'}) 

In [20]:
data_reviews = data_reviews[['asin','date', 'rating']].rename(columns = {'asin': 'ItemId','date': 'ReviewDate', 'rating': 'Rating' })

#### Combining all the three Dataframes into one

In [21]:
data_reviews.head()

ItemId ReviewDate  Rating
43  B001AO4OUC  12-May-09       4
44  B001AO4OUC  19-Nov-12       1
49  B001AO4OUC  25-Nov-12       4
53  B001AO4OUC  18-Oct-09       1
64  B001DCJAJG  22-Jun-14       4

In [22]:
data_Final = data_items.merge(data_reviews, on = ['ItemId','Rating'])
data_Final = data_Final.merge(data_Amazon_Unlocked_New, on = 'Brand Name', how = 'inner')

C:\Users\shweta shukla\Anaconda3\lib\site-packages\pandas\core\reshape\merge.py:1089: UserWarning: You are merging on int and float columns where the float values are not equal to their int representation
  UserWarning,


In [23]:
data_Final.shape

(281, 8)

In [24]:
data_Final.head

<bound method NDFrame.head of          ItemId Brand Name                                           ItemName  \
0    B0096DERAG   Motorola  Motorola MC75A Hand Held Computer Windows Mobi...   
1    B0096QYH80   Motorola  Motorola Droid RAZR M XT907 Verizon Wireless, ...   
2    B0096QYH80   Motorola  Motorola Droid RAZR M XT907 Verizon Wireless, ...   
3    B0096QYH80   Motorola  Motorola Droid RAZR M XT907 Verizon Wireless, ...   
4    B0096QYH80   Motorola  Motorola Droid RAZR M XT907 Verizon Wireless, ...   
..          ...        ...                                                ...   
276  B07K64PTLP     Xiaomi  Xiaomi Mi 8 Pro (128GB, 8GB RAM) with In-Scree...   
277  B07K64PTLP     Xiaomi  Xiaomi Mi 8 Pro (128GB, 8GB RAM) with In-Scree...   
278  B07Y336C8S     Xiaomi  Xiaomi Mi 9 Lite 64GB + 6GB RAM, 6.39” AMOLED ...   
279  B07Y336C8S     Xiaomi  Xiaomi Mi 9 Lite 64GB + 6GB RAM, 6.39” AMOLED ...   
280  B07ZGHYFZ4     Xiaomi  Xiaomi Mi Note 10 128GB 108MP Penta Camera 6.4...  

In [26]:
data_Final.drop("OriginalPrice", axis=1, inplace=True)

In [27]:
data_Final.to_csv("output_ratings.csv", index=False)

In [ ]:
#create table script:

CREATE TABLE Amazon_Review_Ratings(
'ItemId'   VARCHAR (255) NOT NULL,
'BrandName' VARCHAR (255) NULL,
'ItemName' VARCHAR (255)  NULL,
'ItemPrice' NUMERIC (15, 2) NOT NULL,
'OriginalPrice' NUMERIC (15, 2) NULL,
'ReviewDate' date NULL,
'Product Count/Brand'   VARCHAR (255) NULL,
'Rating' DECIMAL (1)
);